### PR TITLE
Add Rec2020 color space

### LINKED ISSUE #20: CSS 4: Predefined color spaces
Will we support all of the CSS 4 predefined color spaces?

See https://www.w3.org/TR/css-color-4/#predefined

### PR DESCRIPTION
**Pull request by tomcur**
Resolves #20, now implementing all CSS Color Module Level 4 color spaces.

Further:

- Add Tom Churchman to AUTHORS list.

### CODE REVIEW

File: color/src/colorspace.rs [hunk 1]
```diff
@@ -421,6 +421,110 @@ impl ColorSpace for ProphotoRgb {
...
+            ],
+            [
+                (-4_638_020_506. / 37_238_079_773.) as f32,
+                (42_187_016_744. / 37_238_079_773.) as f32,
+                (-310_916_465. / 37_238_079_773.) as f32,
+            ],
+            [
+                (-97_469_024. / 5_369_968_309.) as f32,
+                (-3_780_738_464. / 37_589_778_163.) as f32,
+                (42_052_799_795. / 37_589_778_163.) as f32,
```
Comment (tomcur) [inline color/src/colorspace.rs:472]: Some more lily-gilding here. Debatable usefulness, as the transfer function parameters are reals not representable as rationals.

File: color/src/colorspace.rs [hunk 2]
```diff
@@ -421,6 +421,110 @@ impl ColorSpace for ProphotoRgb {
...
+            ],
+        ];
+
+        fn transfer(x: f32) -> f32 {
+            if x.abs() < Rec2020::B * 4.5 {
+                x / 4.5
```
Comment (raphlinus) [inline color/src/colorspace.rs:478]: Should be x * (1.0 / 4.5) for performance.

File: color/src/colorspace.rs [hunk 3]
```diff
@@ -421,6 +421,110 @@ impl ColorSpace for ProphotoRgb {
...
+                (1_772_384_008. / 20_137_682_025.) as f32,
+                (18_035_212_433. / 20_137_682_025.) as f32,
+            ],
+        ];
+
+        fn transfer(x: f32) -> f32 {
+            if x.abs() < Rec2020::B {
+                x * 4.5
+            } else {
+                (Rec2020::A * x.abs().powf(0.45) - Rec2020::A + 1.).copysign(x)
```
Comment (raphlinus) [inline color/src/colorspace.rs:517]: If you group it so it's `(Rec2020::A - 1.)` that's one fewer addition.

**Review by raphlinus — APPROVED** [review]
A couple of picky micro-optimizations, otherwise looks good. Again, spot-checked the color pairs in the test against color.js.

File: color/src/colorspace.rs (hunk 2 above)
Comment (tomcur) [reply color/src/colorspace.rs]: I like how consistently you look for these; I'm learning to remember.

**Comment by raphlinus** [comment]
Btw, we can now delete the `allow` for todos. That could have been done in this PR, but I'll do it in an upcoming one that improves docstrings.

**Comment by DJMcNab** [comment]
Fwiw, that's another key advantage for `expect` - the self-cleansing aspect.

**Pull request merged**